ovh/helpers/helpers.go

```diff
@@ -64,6 +66,19 @@ func ValidateEnum(enum []string) schema.SchemaValidateFunc {
 	}
 }
 
+func ValidateDiagEnum(enum []string) schema.SchemaValidateDiagFunc {
+	return func(v any, p cty.Path) (diags diag.Diagnostics) {
+		err := ValidateStringEnum(v.(string), enum)
+		if err != nil {
+			diags = append(diags, diag.Diagnostic{
```

Comment:
note: you could just use `diags.AddError()`

Reply:
"terraform-plugin-framework/diag" has it not "terraform-plugin-sdk/v2/diag"